src/containers/Application/ApplicationSummary.tsx

```diff
@@ -13,24 +15,51 @@ interface ApplicationProps {
 }
 
 const ApplicationSummary: React.FC<ApplicationProps> = ({ structure }) => {
-  const { replace } = useRouter()
+  const { replace, push } = useRouter()
+  const [shouldRevalidate, setShouldRevalidate] = useState(false)
+  const [isRevalidated, setIsRevalidated] = useState(false)
+  const {
+    userState: { currentUser },
+  } = useUserState()
   const { error, fullStructure } = useGetFullApplicationStructure({
     structure,
+    shouldRevalidate,
```

Comment:
I don't think we need to revalidate ? 

useGetFullApplicationStructure will do full validation first time it fires right ?

Reply:
Yeah, but the intention was to run validation both on page load, and on Submit. You could argue that's overkill, but that's the way it was working in the current version, and there is an outside chance that someone could leave their Summary page open for a long time before submitting, in which time something became invalid.

Reply:
hmm, this would require a full refetch vs just setting shouldRevalidate, right ?

I am ok with approving this one for now, and we can polish of this use case closer to MVP

Reply:
> hmm, this would require a full refetch vs just setting shouldRevalidate, right ?

Setting `shouldValidate` from `false` to `true` forces the hook to re-run, so you get a new fullStructure with validation, which then triggers the rest of the process.

Reply:
but if you had the summary page open in a different tab, you wouldn't get the refetch of current data

Reply:
So you mean, if someone has their Summary page open, and then changes some entries on another device or different tab, then presses "Submit" on the original Summary page, the validation wouldn't be from the new data?

Hmm, seems like a pretty edge case. I guess you could also have a similar problem if someone left their Summary page open and then Submitted from another device -- they'd be able to submit again from the original Summary page. 

The only way round that would be to force the original `useLoadApplication` hook to re-run, not just the `useGetFullStructure` hook (so it updates status, etc.) -- but doing that seems beyond the scope of this issue, don't you think?

Reply:
Yeah, we would need to think this one through (but I think it can wait for now).

Btw I thought my example above was the same as yours below (if now what was the use case for your one?):

> You could argue that's overkill, but that's the way it was working in the current version, and there is an outside chance that someone could leave their Summary page open for a long time before submitting

Reply:
> Btw I thought my example above was the same as yours below (if now what was the use case for your one?):

It's not so much about the application responses changing, more for the case when the existing responses are no longer valid (e.g. name is no longer unique).

Reply:
ahh ok, this case what you've done works nicely